zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +121,74 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() {
+    // Test multiple small transactions to fill a block max size
+
+    // Create a block just below the limit
+    let mut block = generate::large_multi_transaction_block();
+
+    // Serialize the block
+    let mut data = Vec::new();
+    block
+        .zcash_serialize(&mut data)
+        .expect("block should serialize as we are not limiting generation yet");
+
+    assert!(data.len() <= MAX_BLOCK_BYTES as usize);
+
+    // Deserialize by now is ok as we are lower than the limit
+    let block2 = Block::zcash_deserialize(&data[..])
+        .expect("block should deserialize as we are just below limit");
+    assert_eq!(block, block2);
+
+    // Add 1 more transaction to the block, limit will be reached
+    block = generate::oversized_multi_transaction_block();
+
+    // Serialize will still be fine
+    let mut data = Vec::new();
+    block
+        .zcash_serialize(&mut data)
+        .expect("block should serialize as we are not limiting generation yet");
+
+    assert!(data.len() > MAX_BLOCK_BYTES as usize);
+
+    // Deserialize will now fail
+    Block::zcash_deserialize(&data[..]).expect_err("block should not deserialize");
+}
+
+#[test]
+fn block_limits_single_tx() {
+    // Test block limit with a big single transaction
+
+    // Create a block just below the limit
+    let mut block = generate::large_single_transaction_block();
```

Comment:
This is great! It's very easy to read the tests now, and see what they are going to test.